x-pack/plugins/cloud_security_posture/public/components/fleet_extensions/aws_credentials_form.tsx

```diff
@@ -197,11 +197,11 @@ const getInputVarsFields = (
       } as const;
     });
 
-const getDefaultAwsType = (input: Props['input']): AwsCredentialsType =>
-  input.streams[0].vars['aws.credentials.type'].value;
+const getAwsCredentialsType = (input: Props['input']): AwsCredentialsType | undefined =>
+  input.streams[0].vars?.['aws.credentials.type'].value;
 
 export const AwsCredentialsForm = ({ input, newPolicy, updatePolicy }: Props) => {
-  const awsCredentialsType = getDefaultAwsType(input);
+  const awsCredentialsType = getAwsCredentialsType(input) || AWS_CREDENTIALS_OPTIONS[0].id;
```

Comment:
used a default value when there is no value in the stream var. this happens on initial render only. once parent is mounted, we fire an effect to set the value in the policy itself. we could return `null` instead

Reply:
great, i would add a comment to explain that in code. it might be simple but it will clarify why we do it and how the stream vars works